### PR TITLE
Ui updates

### PR DESCRIPTION
**Pull request by RobinRS**
Added Error handling for the connection screen and added a handler for communication about commands on the remote system...

### CODE REVIEW

File: public/src/app/connect/connect.component.html [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        Password
+      </div>
+      <div class="tab" (click)="keyClick($event);" [ngClass]="key ? 'active' : ''">
+        Public Key
+      </div>
+      <div class="tab" (click)="u2fClick($event);" [ngClass]="u2f ? 'active' : ''">
+        U2F
+      </div>
+    </div>
+    <div class="content" [hidden]="!pw">
```
Comment (AlexanderMandera) [inline public/src/app/connect/connect.component.html]: You can also use `*ngIf` to remove the element from the DOM completely when the condition is false.
It is maybe better in this situation.

File: public/src/app/console/console.component.less [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    border-radius: 5px;
+    padding: 25px;
+    font-family: 'Consolas', sans-serif;
+    color: white;
+    box-shadow: 0 2px 4px rgba(0,0,0,.16);
+    position: relative;
+
+    .prompt {
+        display: inline;
+        outline: none;
```
Comment (AlexanderMandera) [inline public/src/app/console/console.component.less]:
```css
min-width: 10px;
word-break: break-all
```

File: public/src/app/console/console.component.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<div class="console">
+    <span id="previouse" style="white-space: pre-line; overflow-y: scroll">{{previouse}}</span>
+    <span id="userServer" contenteditable="false">root@TestServer# </span>
+    <div class="prompt" contenteditable="true" id="prompt" (keydown)="onKey($event)">
```
Comment (AlexanderMandera) [inline public/src/app/console/console.component.html]: For auto-focus and correct wrapping:
`[ngStyle]="{'display': promptText.length > 0 ? 'inline' : 'inline-block'}" #prompt (blur)="prompt.focus()"`

File: public/src/app/console/console.component.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  selector: 'app-console',
+  templateUrl: './console.component.html',
+  styleUrls: ['./console.component.less']
+})
+export class ConsoleComponent implements OnInit {
+
+  lastComments: Array<any>;
+  previouse: String = "";
+  scrollIndex: number = 0;
+  current: Object = {ip: "192.168.2.101", port: 8800, protocol: "SSH"}
```
Comment (AlexanderMandera) [inline public/src/app/console/console.component.ts]: For auto-focus and correct wrapping:
```ts
@ViewChild('prompt') prompt: ElementRef;
promptText: string = "";
```

File: public/src/app/console/console.component.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  constructor(private ipc: IpcService) { }
+
+  ngOnInit() {
+      this.lastComments = [];
```
Comment (AlexanderMandera) [inline public/src/app/console/console.component.ts]: For auto-focus:
```ts
this.prompt.nativeElement.focus();
```

File: public/src/app/console/console.component.ts [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  onKey(event: any) {
+      document.getElementById("prompt").focus();
```
[the remark this answers is not in the document]
Comment (AlexanderMandera) [reply public/src/app/console/console.component.ts]: No focus needed when you type text

[the remark this answers is not in the document]
Comment (AlexanderMandera) [reply public/src/app/console/console.component.ts]:
```ts
this.promptText = this.prompt.nativeElement.innerHTML;
```
Use this for the prompt text (last text entered before the key event).
Also used for make a correct word wrap.

File: public/src/app/console/console.component.html (hunk 3 above)
Comment (AlexanderMandera) [reply public/src/app/console/console.component.html]: You can also use `(focus)="prompt.focus()"` at the `<div class="console">` to focus the prompt every time you click into the console area.

**Comment by RobinRS** [comment]
Fixed mostly everything. Please add the NgModel yourself. I don't understand Angular that good!
(I don't understand it at all 🤔  )

**Pull request merged**